storage/innobase/row/row0mysql.cc

```diff
@@ -1963,6 +1993,49 @@ row_update_for_mysql(row_prebuilt_t* prebuilt)
 	thr->fk_cascade_depth = 0;
 
 run_again:
+	if (vers_set_fields)
+	{
+		/* System Versioning: modify update vector to set
+		   sys_trx_start (or sys_trx_end in case of DELETE)
+		   to current trx_id. */
+		dict_table_t* table = node->table;
+		dict_index_t* clust_index = dict_table_get_first_index(table);
+		upd_t* uvect = node->update;
+		upd_field_t* ufield;
+		dict_col_t* col;
+		unsigned col_idx;
+		if (node->is_delete || vers_delete) {
+			ufield = &uvect->fields[0];
+			uvect->n_fields = 0;
+			node->is_delete = false;
+			node->vers_delete = true;
+			col_idx = table->vers_end;
+		} else {
+			ut_ad(uvect->n_fields < table->n_cols);
+			ufield = &uvect->fields[uvect->n_fields];
+			col_idx = table->vers_start;
+		}
+		col = &table->cols[col_idx];
+		UNIV_MEM_INVALID(ufield, sizeof *ufield);
+		{
+			ulint field_no = dict_col_get_clust_pos(col, clust_index);
```

Comment:
It would be more efficient to have `table->vers_end` and `table->vers_start` to refer to the clustered index fields, rather than table columns.

Reply:
Yes, it is better to have them index field ids. The only place where they used as columns is this:

```c++
const mysql_row_templ_t* t = &prebuilt->mysql_template[table->vers_end];
```